### PR TITLE
enable cmdLineTest_locales on openj9

### PR DESCRIPTION
**Pull request by OscarQQ**
related: backlog issue 384
related: https://github.com/eclipse/openj9/pull/10610

### CODE REVIEW

**Comment by OscarQQ** [comment]
vmfarm build id 458416

**Comment by OscarQQ** [comment]
| Build       | Platform           | Java Version  | JDK Impl |
|:-------------:|:-------------:|:-----:|:-----------:|
|[Grinder/4364](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4364)|ppc64le_linux,ppc64le_linux_xl,x86-64_mac,s390x_linux,s390x_linux_xl,x86-64_linux,ppc64_aix,x86-64_linux_xl|8,11|OpenJ9|

**Comment by OscarQQ** [comment]
@llxia could you review this? thanks

**Comment by pshipton** [comment]
This is not my preference for how to fix the test. It's a loss in testing for IBM Java which does support some (all?) these charsets. The better solution is to modify the test to allow UnsupportedCharsetException without having it cause the test to fail.

**Comment by llxia** [comment]
Yes, I agree. I would suggest making two lists - one for ibm and one for non-ibm for now. And we can work on programmatically detecting supported charsets as an enhancement in the future PR.

**Comment by pshipton** [comment]
I don't think it's hard to fix. Something like the following.
- redirect the output to a file
- record the return code into a variable
- adjust the result based on finding UnsupportedCharsetException in the output
- check the result to determine pass/fail

```
    $1 "${@:2}" &> output ; result = "$?"
    grep UnsupportedCharsetException output 
    if [ "$?" = 0 ]; then
      result = 0
    fi
    if [ $result != 0 ]; then
      echo "bad return code"
    fi
```

**Comment by OscarQQ** [comment]
| Build       | Platform           | Java Version  | JDK Impl |
|:-------------:|:-------------:|:-----:|:-----------:|
|Grinder/11714|ppc64le_linux,ppc64le_linux_xl,x86-64_mac,s390x_linux,s390x_linux_xl,x86-64_linux,ppc64_aix,x86-64_linux_xl|8,11|OpenJ9|

**Comment by OscarQQ** [comment]
@llxia could you review this? thanks :)

**Review by pshipton — APPROVED** [review]

File: test/functional/cmdLineTests/locales/run_all_locales.sh [hunk 1]
```diff
@@ -51,9 +51,13 @@ do
     export LANG=$locale;
     
     # run first argument and pass remaining arguments
-    $1 "${@:2}";
-    
-    if [ "$?" != 0 ]; then
+    output=$($1 "${@:2}" 2>&1) ; result="$?"
+    echo $output
+    grep 'UnsupportedCharsetException' <<< $output 
+    if [ "$?" = 0 ]; then
```
Comment (keithc-ca) [inline test/functional/cmdLineTests/locales/run_all_locales.sh:57]: These two lines should be combined and grep output suppressed:
```
if grep -q UnsupportedCharsetException <<< "$output" ; then
```
(note also that `UnsupportedCharsetException` need not be quoted and that `$output` should be quoted)

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Review by keithc-ca — APPROVED** [review]

**Comment by llxia** [comment]
Please update the above Ginder links and run a vmfarm personal build. Thanks

**Comment by OscarQQ** [comment]
| Build       | Platform           | Java Version  | JDK Impl |
|:-------------:|:-------------:|:-----:|:-----------:|
|Grinder/11857|ppc64le_linux,ppc64le_linux_xl,s390x_linux,s390x_linux_xl,x86-64_linux,ppc64_aix,x86-64_linux_xl|8,11|OpenJ9|
|[Grinder/4538](https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4538)|x86-64_mac|8,11|OpenJ9|

vmfarm build_id=458927

**Comment by llxia** [comment]
jenkins test extended all jdk8,jdk11

**Comment by llxia** [comment]
jenkins test extended win jdk8

**Comment by llxia** [comment]
PR builds failed in [Build_JDK8_x86-64_windows](https://ci.eclipse.org/openj9/job/Build_JDK8_x86-64_windows_Personal/946/console). This failure is not related to this PR. Since the test passed on all other platforms and the test does not run on windows, I am going to merge this PR.

**Review by llxia — APPROVED** [review]
LGTM

**Pull request merged**